core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -86,7 +86,13 @@ class SnappyStoreHiveCatalog(externalCatalog: SnappyExternalCatalog,
 
   val sparkConf: SparkConf = snappySession.sparkContext.getConf
 
-  private[sql] var client = metadataHive
+  private var _client = metadataHive
+
+  private[sql] def client = {
+    // check initialized meta-store (including initial consistency check)
+    Misc.getMemStoreBooting.getExternalCatalog
```

Comment:
Any special reason to force initialize here? First access anyways initializes it.

Reply:
Primarily to ensure that catalog consistency check is complete before any catalog reads/writes. The catalog consistency check has been moved to background task as part of these changes and don't block the startup, so on first use the check has to ensured complete (or else throw an exception if there is still some unexpected inconsistency that could not be repaired).